From: "Yan, Zheng" <zheng.z.yan@intel.com>
Date: Mon, 24 Mar 2014 19:15:05 +0800
Subject: [PATCH] ceph: don't include ceph.{file,dir}.layout vxattr in
 listxattr()
Git-commit: cc48c3e85f7fc48092f2e9874f1a07dd997d9184
Patch-mainline: v3.15
References: Fate#318586


This avoids 'cp -a' modifying layout of new files/directories.

Signed-off-by: Yan, Zheng <zheng.z.yan@intel.com>
Acked-by: NeilBrown <neilb@suse.com>

---
 fs/ceph/xattr.c |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

--- a/fs/ceph/xattr.c
+++ b/fs/ceph/xattr.c
@@ -227,7 +227,7 @@ static struct ceph_vxattr ceph_dir_vxatt
 		.name_size = sizeof("ceph.dir.layout"),
 		.getxattr_cb = ceph_vxattrcb_layout,
 		.readonly = false,
-		.hidden = false,
+		.hidden = true,
 		.exists_cb = ceph_vxattrcb_layout_exists,
 	},
 	XATTR_LAYOUT_FIELD(dir, layout, stripe_unit),
@@ -254,7 +254,7 @@ static struct ceph_vxattr ceph_file_vxat
 		.name_size = sizeof("ceph.file.layout"),
 		.getxattr_cb = ceph_vxattrcb_layout,
 		.readonly = false,
-		.hidden = false,
+		.hidden = true,
 		.exists_cb = ceph_vxattrcb_layout_exists,
 	},
 	XATTR_LAYOUT_FIELD(file, layout, stripe_unit),
